### PR TITLE
🚧 feat: add support for vmrest

### LINKED ISSUE #157: Add support for the VMware Workstation Pro API with the vmware-vmx builder
#### Community Note

Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request.
If you are interested in working on this issue or have submitted a pull request, please leave a comment.

#### Description

Create a new "driver" for `vmrest`, the API that ships with VMWare Workstation Pro. Due to API limitations, only the vmware-vmx builder can be implemented. 

#### Use Case(s)

Implement the vmware-vmx functionality using the API. Enables using Packer from a remote host or from a container.

#### Potential configuration

Already implemented and **ready for a PR**. See https://github.com/niwamo/packer-plugin-vmware.

I also updated /docs/builders/vmx.mdx and ran `.web-docs/scripts/compile-to-webdocs.sh`.

Opening an issue for discussion prior to making the PR, as directed in the README.

The code is reasonably well-commented, and I endeavored to limit changes outside of the new driver file to the greatest extent possible. Please take a look and lmk if/when you're ready for a PR.

#### Potential References

If you'd like to repeat my tests:
- Set the API credentials on the VMWare host
- Run the API
- Install the updated plugin in a WSL container
- Create a new VM using the "New Virtual Machine" menu in VMWare
   - Assign a Debian .iso for the installer, but don't run the VM
- Use the created .vmx filepath in a configuration like the following:

```hcl
packer {
  required_plugins {
    vmware = {
      version = ">= 1.0"
      source = "github.com/niwamo/vmware"
    }
  }
}

source "vmware-vmx" "vmrest" {
  remote_type     = "vmrest"
  remote_host     = // the VMWare host's WSL interface IP
  remote_port     = 8697
  remote_username = // username you set for the API
  remote_password = // password you set for the API
  skip_export     = true

  source_path = // vmx filepath - make sure all backslashes are doubled/escaped, e.g. "C:\\Users\\test.vmx"

  vnc_bind_address  = "0.0.0.0"

  http_content = {
    "/preseed.cfg" = file("./preseed.cfg")
  }

  boot_wait    = "10s"
  boot_command = [
    "<esc><wait>",
    "auto url=http://{{ .HTTPIP }}:{{ .HTTPPort }}/preseed.cfg",
    "<enter>"
  ]

  ssh_username      = // user created by the preseed
  ssh_password      = // pwd for user created by the preseed
  ssh_timeout       = "30m"
  shutdown_command  = "shutdown -P now"
}

build {
  sources = ["sources.vmware-vmx.vmrest"]
  provisioner "shell" {
    script = "./scripts/test-1.sh"
  }
}
```

Note that this test config requires 
- a valid "preseed.cfg" to be present
- a valid "./scripts/test-1.sh" to be present
- your Windows firewall to allow all inbound connections from your WSL distro

### PR DESCRIPTION
**Pull request by niwamo**
As described in #157:

Creates a new driver for vmrest, the API that ships with VMWare Workstation Pro. Due to limitations in the API, only the vmware-vmx builder can be implemented. 

Checks have been added to the vmware-iso builder to throw an explanatory error if a user mistakenly tries to use the vmrest remote_type. Additional checks have been added to throw errors if incompatible configurations are set while using the vmrest remote_type. 

The vmware-vmx documentation was also updated. 

A test for the new driver has been provided in #157.

Closes #157

### CODE REVIEW

**Comment by niwamo** [comment]
All issues identified by the first round of checks should be addressed. There were three categories of issues:

1. Tests for driver_config.go that I failed to update, including an error message that was intentionally changed, and the adjustment to only set esxi-relevant defaults if esxi is selected; Both have been fixed
2. A small number of linter-identified issues in driver_vmrest.go. All good catches; All now addressed.
3. Exactly two lines with discrepancies between docs and web-docs; Now fixed

I've also re-run my live test of the vmrest builder to ensure changes did not affect functionality

File: .web-docs/components/builder/vmx/README.md [hunk 1]
```diff
@@ -1132,7 +1140,65 @@ Since ovftool is only capable of password based authentication
 - `vnc_disable_password` - This must be set to "true" when using VNC with
 ESXi 6.5 or 6.7.
 
-### SSH key pair automation
+### Building with the VMWare Workstation Pro API (vmrest)
+
+VMWare Workstation Pro ships with a 'vmrest' executable enabling programmatic
```
Comment (tenthirtyam) [inline .web-docs/components/builder/vmx/README.md:1145]: "VMware" for all occurrences of "VMWare", please.

Comment (niwamo) [reply .web-docs/components/builder/vmx/README.md]: Thank you for the review! I've corrected "VMWare" to "VMware" in all instances where I wrote the former. 

(Note: I did not make the change for instances of "VMWare" that preexisted my contribution)

**Comment by niwamo** [comment]
@tenthirtyam Since you were kind enough to review - do you know if there are additional steps I should be taking and/or reviewers I should be talking to? This has been slow to take off, and I want to make sure I'm doing my due diligence. The members of my team that need this functionality are already using a fork, but it would be nice to have this in the official plugin.

**Comment by nywilken** [comment]
@tenthirtyam we will throw some time on your calendar to sync up on this change before starting to review.

**Comment by niwamo** [comment]
@tenthirtyam Thanks for the recent follow-up(s). Going to close this merge request, re-factor/clean-up the code, and re-open a request in the next couple of weeks. Will leave the same issue open. Thanks.

**Comment by tenthirtyam** [comment]
HI @niwamo 👋 - sorry. I've been meaning to get back with you on this one.

I'm in the midst of doing some consolidation of the drivers based on the recent changes the desktop hypervisors (personal license, too). Those are moving along well and we'll plan to pull out Player around the next major release for Workstation/Fusion and Player EOL.

With regards to `vmrest`, I'd suggest pausing on this one for the time being. There are some product enhancements planned (can't share publically, sorry!) where there will be some changes that may make these changes moot but lead to a more streamlined approach.

**Ryan Johnson**
Distinguished Engineer, VMware by Broadcom

**Pull request closed without merging**